### PR TITLE
Remove `public_key` from `Tree`.

### PR DESCRIPTION
**Pull request by pphaneuf**
Part of #2450.

### Checklist

- [ ] I have updated the [CHANGELOG](CHANGELOG.md).
- [ ] I have updated [documentation](docs/) accordingly (including the [feature implementation matrix](docs/Feature_Implementation_Matrix.md)).

### CODE REVIEW

File: storage/mysql/admin_storage.go [hunk 1]
```diff
@@ -301,7 +301,7 @@ func (t *adminTX) CreateTree(ctx context.Context, tree *trillian.Tree) (*trillia
 		nowMillis,
 		nowMillis,
 		privateKey,
-		newTree.PublicKey.GetDer(),
+		[]byte{}, // Unused, filling in for backward compatibility.
```
Comment (pavelkalinnikov) [inline storage/mysql/admin_storage.go:304]: Should this and other constants be moved directly to the SQL statement, instead of being added as parameters?

**Review by pavelkalinnikov — APPROVED** [review]

File: storage/mysql/admin_storage.go (hunk 1 above)
Comment (pphaneuf) [reply storage/mysql/admin_storage.go:304]: I did something similar to this in the previous PRs, where I'm trying to touch this as little as possible, and I'd try a separate PR to tweak the SQL statements?

Comment (pavelkalinnikov) [reply storage/mysql/admin_storage.go:304]: Yep, separate PR sounds good.

**Pull request merged**